test/validate_resnet50/validation.py

```diff
@@ -0,0 +1,263 @@
+        print("Running: ", validate_with_real_data_command_TF)
+        output, error_output = command_executor(
+            validate_with_real_data_command_TF)
+        output_string = str(output, 'utf-8')
+
+        if output:
+            reference_outputs_total_loss, reference_outputs_top1_acc, reference_outputs_top5_acc = parse_training_output(
+                output_string)
+        elif error_output:
+            print("Something went wrong executing the command ",
+                  validate_with_real_data_command_NG)
+            print(str(error_output, 'utf-8'))
+            exit(1)
+
+        write_to_file(
+            "resnet50_validaionResultReference" + str(batch_size) + ".txt",
+            output_string)
+
+    print("reference total loss ", reference_outputs_total_loss)
+    print("reference top1Acc ", reference_outputs_top1_acc)
+    print("reference top5Acc ", reference_outputs_top5_acc)
+
+    # Compare the TF output and NG output
+    # TF CPU results and GPU results are not the same, so for TF results
+    # Need to run with TF GPU
+    assert len(ngraph_outputs_total_loss) == len(
+        reference_outputs_total_loss), "Number of total_loss values mismatch"
+    assert len(ngraph_outputs_top1_acc) == len(
+        reference_outputs_top1_acc), "Number of top1_accuracy values mismatch"
+    assert len(ngraph_outputs_top5_acc) == len(
+        reference_outputs_top5_acc), "Number of top5_accuracy values mismatch"
+
+    loss_norms = calculate_norm_values(ngraph_outputs_total_loss,
+                                       reference_outputs_total_loss)
+    top1Acc_norms = calculate_norm_values(ngraph_outputs_top1_acc,
+                                          reference_outputs_top1_acc)
+    top5Acc_norms = calculate_norm_values(ngraph_outputs_top5_acc,
+                                          reference_outputs_top5_acc)
+
+    print(
+        "loss norms are %f %f %f " %
+        (loss_norms["l1_norm"], loss_norms["l2_norm"], loss_norms["inf_norm"]))
+    print("top1Acc norms are %f %f %f " %
+          (top1Acc_norms["l1_norm"], top1Acc_norms["l2_norm"],
+           top1Acc_norms["inf_norm"]))
+    print("top5Acc norms are %f %f %f " %
+          (top5Acc_norms["l1_norm"], top5Acc_norms["l2_norm"],
+           top5Acc_norms["inf_norm"]))
+
+    loss_result = check_validation_results(loss_norms, "total_loss")
+    top1Acc_result = check_validation_results(loss_norms, "top1 Accuracy")
+    top5Acc_result = check_validation_results(loss_norms, "top5 Accuracy")
+
+    if ((loss_result and top1Acc_result and top5Acc_result)):
+        print("Validation test pass")
+
+    # reapply the patch
+    output, error_output = command_executor(
+        'git apply -R ' + './datasets_make_deterministic.patch')
+
+
+# Validation with synthetic data
+
+if __name__ == "__main__":
+    reference_file_name_realData = ''
+    reference_file_name_syntheticData = ''
```

Comment:
we are sending in empty strings, and counting on the `run_validation` to detect that and create new files if needed, right?

Reply:
yes.